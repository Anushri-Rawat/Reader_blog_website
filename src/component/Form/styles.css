.form-input {
  width: 100%;
  padding: 10px;
  border: 1px solid #ccc;
  resize: vertical;
}
label {
  font-weight: 500;
  font-size: 18px;
}
