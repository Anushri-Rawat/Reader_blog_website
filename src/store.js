import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import posts from "./reducers";

const initialState = {
  loading: false,
  posts: JSON.parse(localStorage.getItem("posts")) || [],
};

const store = () =>
  createStore(posts, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
