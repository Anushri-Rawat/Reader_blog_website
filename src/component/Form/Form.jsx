import React, { useState } from "react";
import "./styles.css";
import { useDispatch, useSelector } from "react-redux";

const Form = () => {
  const dispatch = useDispatch();
  const { posts } = useSelector((state) => state);
  const [userId, setUserId] = useState(1);
  const [title, setTitle] = useState("");
  const [body, setBody] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    dispatch({
      type: "ADD_POST",
      payload: { userId, id: posts.length, title, body },
    });
  };

  return (
    <form
      className="d-flex flex-column gap-3"
      data-testid="form"
      onSubmit={submitHandler}
    >
      <h2 className="title mt-0">Add Blog</h2>
      <div className="row align-items-center">
        <div className="col-md-3">
          <label htmlFor="userId">User Id</label>
        </div>
        <div className="col-md-9">
          <select
            name="UserId"
            id="userId"
            className="form-input"
            onChange={(e) => setUserId(e.target.value)}
          >
            {Array.from({ length: 10 }, (_, i) => i + 1).map((item, i) => (
              <option value={item} key={i}>
                {item}
              </option>
            ))}
          </select>
        </div>
      </div>
      <div className="row align-items-center">
        <div className="col-md-3">
          <label htmlFor="title">Title</label>
        </div>
        <div className="col-md-9">
          <input
            type="text"
            id="title"
            placeholder="Title"
            className="form-input"
            value={title}
            onChange={(e) => setTitle(e.target.value)}
          />
        </div>
      </div>
      <div className="row align-items-center">
        <div className="col-md-3">
          <label htmlFor="body">Body</label>
        </div>
        <div className="col-md-9">
          <textarea
            type="text"
            id="body"
            rows={5}
            placeholder="Body"
            className="form-input"
            value={body}
            onChange={(e) => setBody(e.target.value)}
          />
        </div>
      </div>
      <button className="btn btn-dark rounded-0">Add Blog</button>
    </form>
  );
};

export default Form;
