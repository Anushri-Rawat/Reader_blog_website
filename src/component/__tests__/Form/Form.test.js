import { render as rtlRender, screen, cleanup } from "@testing-library/react";
import Form from "../../Form/Form";
import store from "../../../store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

const render = (component) =>
  rtlRender(<Provider store={store()}>{component}</Provider>);

test("should render form component", () => {
  render(<Form />);
  const formElement = screen.getByTestId("form");
  expect(formElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <Form />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
