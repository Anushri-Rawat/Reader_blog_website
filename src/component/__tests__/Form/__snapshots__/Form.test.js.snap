// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="d-flex flex-column gap-3"
  data-testid="form"
  onSubmit={[Function]}
>
  <h2
    className="title mt-0"
  >
    Add Blog
  </h2>
  <div
    className="row align-items-center"
  >
    <div
      className="col-md-3"
    >
      <label
        htmlFor="userId"
      >
        User Id
      </label>
    </div>
    <div
      className="col-md-9"
    >
      <select
        className="form-input"
        id="userId"
        name="UserId"
        onChange={[Function]}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
        <option
          value={6}
        >
          6
        </option>
        <option
          value={7}
        >
          7
        </option>
        <option
          value={8}
        >
          8
        </option>
        <option
          value={9}
        >
          9
        </option>
        <option
          value={10}
        >
          10
        </option>
      </select>
    </div>
  </div>
  <div
    className="row align-items-center"
  >
    <div
      className="col-md-3"
    >
      <label
        htmlFor="title"
      >
        Title
      </label>
    </div>
    <div
      className="col-md-9"
    >
      <input
        className="form-input"
        id="title"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="row align-items-center"
  >
    <div
      className="col-md-3"
    >
      <label
        htmlFor="body"
      >
        Body
      </label>
    </div>
    <div
      className="col-md-9"
    >
      <textarea
        className="form-input"
        id="body"
        onChange={[Function]}
        placeholder="Body"
        rows={5}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="btn btn-dark rounded-0"
  >
    Add Blog
  </button>
</form>
`;
